SalesforceCommerceCloud/pwa-kit
# [App Extensibility ⚙️] Solve IDE wildcard import resolution (@W-16825282@)

# Description



### Problem

The IDEs are unable to resolve the overridable imports identified using a wildcard correctly and provide the proper navigation and IntelliSense.

This results in errors with red squiggly lines below the wildcard import syntax.

Example
```
import withRedBorder from '*/components/with-red-border'
```
Error 
```
TS2307: Cannot find module '*/components/with-red-border'.
```

### Solution

The PR implements two changes:
1. Replace the wildcard import symbol `*` with the dollar `override!` symbol to comply with the ts `compilerOptions.path` pattern limitation of using at most one '*' character. See [ts config wildcard substitutions docs](https://www.typescriptlang.org/docs/handbook/modules/reference.html#wildcard-substitutions).
2. Add a `tsconfig.json` file with the `compilerOption.path` mapping the paths starting with `override!/*` to `src/*`.  

```
{
    "compilerOptions": {
        ...
        "paths": {
            "override!/*": [
                "src/*",
            ]
        },
...
```

Using a different symbol to identify overridable imports solves the error caused by the wildcard symbol.

```
npx tsc --build
tsconfig.json:12:13 - error TS5061: Pattern '*/*' can have at most one '*' character.

12             "*/*": [
               ~~~~~


Found 1 error.
```

# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- Replace the wildcard import symbol `*` with the dollar `override!`.
- Adds a `tsconfig.json` file. 

# How to Test-Drive This PR

- Checkout this branch `solve-wildcard-imports` in your IDE.
- Open the `packages/extension-base/src/setup-app.ts` file.
- Verify:
    1. There are no red squiggly lines below the dollar imports.
    2. The IDE navigates to the expected file when clicking (Cmd+ Click) on the dollar import.

# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
